java-checks/src/main/java/org/sonar/java/closeresource/CloseableVisitor.java

```diff
@@ -116,26 +110,23 @@ protected boolean isSymbolRelevant(Symbol symbol) {
   }
 
   @Override
-  public void visitVariable(VariableTree tree) {
-    super.visitVariable(tree);
-    ExpressionTree initializer = tree.initializer();
-
+  protected void onAssignment(ExecutionState executionState, Tree tree, Symbol variable, ExpressionTree expression) {
     // check first usage of closeables in order to manage use of same symbol
-    ignoreClosableSymbols(initializer);
-
-    Symbol symbol = tree.symbol();
-    if (isCloseableOrAutoCloseableSubtype(symbol.type())) {
-      executionState.markValueAs(symbol, getCloseableStateFromExpression(symbol, initializer));
+    ignoreClosableSymbols(executionState, expression);
+    if (tree.is(Tree.Kind.VARIABLE) || ((AssignmentExpressionTree) tree).variable().is(Tree.Kind.IDENTIFIER, Tree.Kind.MEMBER_SELECT)) {
+      if (isCloseableOrAutoCloseableSubtype(variable.type())) {
```

Comment:
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Merge this if statement with the enclosing one.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1066)